public/javascript/pump/socket.js

```diff
@@ -134,9 +149,34 @@
             }
         };
 
-        sock.onclose = function() {
-            // XXX: reconnect?
+        sock.onclose = function(e) {
             Pump.socket = null;
+            // Reconnect on broken connections, not for normal close
+            if (e.code !== 1000) {
+                sock.reconnect(e);
+            }
+        };
+
+        sock.reconnect = function(e) {
+            _socket.retryAttempts++;
+
+            if (_socket.retryTimer) {
+                clearTimeout(_socket.retryTimer);
+                _socket.retryTimer = null;
```

Comment:
I'm not _super_ familiar with the `clearTimeout` API but I'm pretty sure this is redundant?

Reply:
Why redundant?

Reply:
@vxcamiloxv lol. Because I was super dumb and didn't read the docs. For some reason I thought `clearTimeout` set the value to null or something, but honestly that makes absolutely no sense so I don't know what I was thinking. This looks fine.